src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,198 @@
+/// </summary>
+public class SegmentedBar : ProgressBar
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+
+    public void BarToggled(InputEvent @event, IconProgressBar bar)
+    {
+        if (@event is InputEventMouseButton eventMouse && @event.IsPressed())
+        {
+            IconBarConfig config = new IconBarConfig(bar);
+            config.Disabled = !config.Disabled;
+            new SegmentedBarConfig(this).handleBarDisabling(bar);
+        }
+    }
+}
+
+/// <summary>
+///   Configuration class for SegmentedBar class
+/// </summary>
+public class SegmentedBarConfig
+{
+    private SegmentedBar target;
+
+    public SegmentedBarConfig(SegmentedBar target)
+    {
+        this.target = target;
+    }
+
+    public float[] Size
+    {
+        get { return new float[]{ target.RectSize.x, target.RectSize.y }; }
+        set
+        {
+            target.RectSize = new Vector2(value[0], value[1]);
+            target.RectMinSize = target.RectSize;
+        }
+    }
+
+    public string Type
+    {
+        get { return target.Type; }
+        set { target.Type = value; }
+    }
+
+    public void updateAndMoveBars(Dictionary<string, float> data)
+    {
+        removeUnusedBars(target, data);
+
+        int location = 0;
+        foreach (var process in data)
+        {
+            createAndUpdateProcessBar(process, target, location);
+            location++;
+        }
+
+        foreach (var process in data)
+        {
+            if (target.HasNode(process.Key))
+                updateDisabledBars(process, target);
+        }
+
+        foreach (var process in data)
+        {
+            if (target.HasNode(process.Key))
+                moveBars(target.GetNode<IconProgressBar>(process.Key));
+        }
+    }
+
+    private void removeUnusedBars(ProgressBar parent, Dictionary<string, float> processes)
+    {
+        foreach (IconProgressBar progressBar in parent.GetChildren())
+        {
+            bool match = false;
+            foreach (var process in processes)
+            {
+                if (progressBar.Name == process.Key)
+                    match = true;
+            }
+
+            if (!match)
+                progressBar.Free();
+        }
+    }
+
+    private void createAndUpdateProcessBar(KeyValuePair <string, float> process, ProgressBar parent, int location = -1)
+    {
+        if (parent.HasNode(process.Key))
+        {
+            IconProgressBar progressBar = (IconProgressBar)parent.GetNode(process.Key);
+            IconBarConfig config = new IconBarConfig(progressBar);
+            if (config.Disabled) return;
+            config.LeftShift = getPreviousBar(parent, progressBar).RectSize.x + getPreviousBar(parent, progressBar).MarginLeft;
+            config.Size = new Vector2((float)Math.Floor(process.Value / parent.MaxValue * 318), 30);
```

Comment:
Why there is 318 here? That's a magic value. And I also think that that probably should also be configurable to fit the bars in other places.